src/Psy/Shell.php

```diff
@@ -71,6 +73,11 @@ public function __construct(Configuration $config = null)
         $this->context  = new Context();
         $this->includes = array();
         $this->readline = $this->config->getReadline();
+        if ($this->config->getTabCompletion()) {
+            $this->completion = new TabCompletion();
+            $this->completion->setContext($this->context);
```

Comment:
I was going to comment that I'd prefer injecting dependencies via the constructor, but it looks like you've already written it that way :) Let's use that here?